src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1700,28 +1701,30 @@ private void CalculateIORedirection(bool isWindowsApplication, out bool redirect
             "Performance",
             "CA1822:Mark members as static",
             Justification = "Accesses instance members in preprocessor branch.")]
+        [SkipLocalsInit]
         private bool IsExecutable(string path)
         {
 #if UNIX
             return Platform.NonWindowsIsExecutable(this.Path);
 #else
 
             string myExtension = System.IO.Path.GetExtension(path);
+            string pathext = Environment.GetEnvironmentVariable("PATHEXT");
 
-            var pathext = Environment.GetEnvironmentVariable("PATHEXT");
-            string[] extensionList;
-            if (string.IsNullOrEmpty(pathext))
-            {
-                extensionList = new string[] { ".exe", ".com", ".bat", ".cmd" };
-            }
-            else
-            {
-                extensionList = pathext.Split(';');
-            }
+            ReadOnlySpan<char> extensionList = string.IsNullOrEmpty(pathext)
+                ? ".exe;.com;.bat;.cmd"
+                : pathext;
+
+            int extensionCount = extensionList.Count(';') + 1;
+            Span<Range> extensions = extensionCount < 0x20
+                ? (stackalloc Range[0x20]).Slice(0, extensionCount)
+                : new Range[extensionCount];
```

Comment:
Please use named constatnt for 0x20 like `const int MaxStackBuffer = 256;`

(And if you care about performance/best code generation you could use const stack alloc buffer explicitly.)

Reply:
> (And if you care about performance/best code generation you could use const stack alloc buffer explicitly.)

I'm just going to revert that change as it's unrelated and don't want it to hold up the PR.

That said, can you give an example of what you mean here? I am using a fixed sized stackalloc buffer. If you mean use stackalloc and never use an array, we can't be sure that's reasonable given we're getting user input in the form of a environment variable

Reply:
> That said, can you give an example of what you mean here?

If I remember correctly Roslyn optimizes  stackalloc with const size moving stack allocation before function call. If the stackalloc is under condition the optimization doesn't work and Roslyn has to generate more complex dynamic code for stack allocation and deallocation.
```c#
var arr = stackalloc string[256];
if (we_need > 256)
{
    arr = new string[we_need];
}
else
{
    arr = arr.Slice(we_need);
}

Reply:
> If I remember correctly Roslyn optimizes stackalloc with const size moving stack allocation before function call. If the stackalloc is under condition the optimization doesn't work and Roslyn has to generate more complex dynamic code for stack allocation and deallocation.

Ah alright then we're thinking of the same thing. It's doing `stackalloc char[something.Length]` that's bad. You want to allocate the max you're willing to and then just slice it afterwards. That's what my code did before I reverted.

The array part doesn't matter, it's just variable sized `stackalloc` that has a large perf hit